graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -268,47 +304,128 @@ def enter(self, node, key, parent, path, ancestors):
+        self, node: ObjectTypeDefinitionNode, key, parent, path, ancestors
+    ):
         """If the node's name matches the query type, record that we left the query type."""
+        if not node.fields:
+            raise CascadingSuppressionError(
+                f"Type renamings {self.renamings} suppressed every type in the schema so it will be"
+                f" impossible to query for anything. To fix this, check why the `renamings` "
+                f"argument of `rename_schema` mapped every type to None. Note that adding "
+                f"suppressions may lead to other types, fields, unions, etc. requiring suppression "
+                f"so you may need to iterate on this before getting a legal schema."
+            )
         if node.name.value == self.query_type:
             self.in_query_type = False
 
-    def enter_field_definition(self, node, *args):
+    def enter_field_definition(self, node: FieldDefinitionNode, *args):
         """If inside the query type, rename field and add the name pair to reverse_field_map."""
         if self.in_query_type:
             field_name = node.name.value
             new_field_name = self.renamings.get(field_name, field_name)  # Default use original
             if new_field_name == field_name:
                 return None
+            if new_field_name is None:
+                # suppress the type
+                return REMOVE
             else:  # Make copy of node with the changed name, return the copy
                 field_node_with_new_name = get_copy_of_node_with_new_name(node, new_field_name)
                 return field_node_with_new_name
+
+
+class CascadingSuppressionCheckVisitor(Visitor):
+    def __init__(self, renamings: Dict[str, Optional[str]], query_type: str):
+        """Create a visitor to check that suppression does not cause an illegal state.
+
+        Args:
+            renamings: Dict[str, Optional[str]], from original field name to renamed field name. Any
+                       name not in the dict will be unchanged
+            query_type: str, name of the query type (e.g. RootSchemaQuery)
+        """
+        # Note that as field names and type names have been confirmed to match up, any renamed
+        # query type field already has a corresponding renamed type. If no errors, due to either
+        # invalid names or name conflicts, were raised when renaming type, no errors will occur when
+        # renaming query type fields.
+        self.in_query_type = False
+        self.renamings = renamings
+        self.query_type = query_type
+
+    def enter_object_type_definition(self, node: ObjectTypeDefinitionNode, *args):
+        """If the node's name matches the query type, record that we entered the query type."""
+        if node.name.value == self.query_type:
+            self.in_query_type = True
+
+    def leave_object_type_definition(
+        self, node: ObjectTypeDefinitionNode, key, parent, path, ancestors
+    ):
```

Comment:
Remember to complete type hints on all functions, including `-> None` if the function doesn't return anything. Also remember that `__init__` does not return anything, so it should always be typed `-> None`.